dotcom-rendering/src/lib/discussionApi.tsx

```diff
@@ -176,42 +174,41 @@ export const reply =
 		data.append('body', body);
 		const authOptions = getOptionsHeadersWithOkta(authStatus);
 
-		return fetch(url, {
+		const resp = await fetch(url, {
 			method: 'POST',
 			body: data.toString(),
 			headers: {
 				'Content-Type': 'application/x-www-form-urlencoded',
 				...options.headers,
-				...(authOptions.headers !== undefined
-					? authOptions.headers
-					: {}),
+				...authOptions.headers,
 			},
 			credentials: authOptions.credentials,
-		}).then((resp) => resp.json());
+		});
+
+		return await resp.json();
 	};
 
 //todo: come back and parse the response properly and set a proper return type for the error case
-export const getPicks = (
+export const getPicks = async (
 	shortUrl: string,
 ): Promise<CommentType[] | undefined> => {
 	const url =
 		joinUrl(options.baseUrl, 'discussion', shortUrl, 'topcomments') +
 		objAsParams(defaultParams);
 
-	return (
-		fetch(url, {
-			headers: {
-				...options.headers,
-			},
-		})
-			.then((resp) => resp.json())
-			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-			.then((json) => json.discussion.comments)
-			.catch((error) => console.error(`Error fetching ${url}`, error))
-	);
+	const resp = await fetch(url, {
+		headers: {
+			...options.headers,
+		},
+	});
+
+	const json = await resp.json();
+
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- YOLO
+	return json.discussion.comments;
```

Comment:
How do people feel about a little parsing?

I'd be keen to look at [Valibot](https://valibot.dev/) or [Zod](https://zod.dev/)

Reply:
Parsing is 👍.